### DISCUSSION TITLE
Deliver doesn't create an appstore version

### DISCUSSION THREAD

**Discussion by SolorzanoJose** [fastlane/fastlane#20255] [General]
I'm trying to use Deliver to submit for appstore review but it's not creating an appstore version so I get the following error:\

`Cannot submit for review - could not find an editable version for 'IOS'`

My code is as follows:

`    deliver(
        app_identifier: appId,
        api_key_path: apiPath,
        submit_for_review: true,
        automatic_release: true,
        force: true, # Skip HTMl report verification
        skip_metadata: true,
        skip_screenshots: true,
        skip_binary_upload: true,
        run_precheck_before_submit: false
    )`

Do I need to run another command to create a version?

**Comment by lucgrabowski** [comment]
Have you uploaded a build that you want to submit to review with `pilot`? Uploading a build with`pilot` won't create a new editable version on Appstore Connect  (if you login and check your app, on left hand side you'll see a title "iOS app" with a + button that allows to create a new version of the app and below you'll see a currently released version of the app).

As`skip_binary_upload: true` in deliver call, I think that you can:
1. add `app_version` parameter to the call as written in table in docs https://docs.fastlane.tools/actions/deliver/
2. upload a new build with `deliver` and it will create a new editable version of the app
or as a last resort
3. you can create a new version in AppstoreConnect manually

**Comment by SolorzanoJose** [comment]
@lucgrabowski I tried number 1 but that doesn't create a version, it instead complains about the version not existing

**Reply by lucgrabowski** [reply]
I've checked code briefly and it looked like 1 should work.
And what if you set `app_version`, `skip_binary_upload: true` and e.g. try uploading metadata with `skip_metadata: false` ?

**Comment by DoodySq** [comment]
Your article content is being very much interested, I am very impressed with your post. I hope to receive more great posts.


[BenefitWallet](https://www.benefitwallet.me/)
